helm/artifactory-ha-values.yaml

```diff
@@ -3,32 +3,39 @@ global:
     observability:
       branch: master
   prometheus:
-    loki_url: http:\/\/loki.jfrog-plg:3100
+    loki_url: http://loki.jfrog-plg:3100
     release_name: prometheus
 installerInfo: '{ "productId": "Helm_prometheus_artifactory/{{ .Chart.Version }}", "features": [ { "featureId": "ArtifactoryVersion/{{ default .Chart.AppVersion .Values.artifactory.image.version }}" }, { "featureId": "{{ if .Values.postgresql.enabled }}postgresql{{ else }}{{ .Values.database.type }}{{ end }}/0.0.0" }, { "featureId": "Platform/{{ default "kubernetes" .Values.installer.platform }}" },  { "featureId": "Channel/Helm_prometheus_artifactory" } ] }'
 artifactory:
   openMetrics:
     enabled: true
+  customInitContainersBegin: |
+    - name: "prepare-fluentd-conf-on-persistent-volume"
+      image: "{{ .Values.initContainerImage }}"
+      imagePullPolicy: "{{ .Values.artifactory.image.pullPolicy }}"
+      command:
+        - 'sh'
+        - '-c'
+        - >
+          mkdir -p {{ .Values.artifactory.persistence.mountPath }}/etc/fluentd/;
+          curl https://raw.githubusercontent.com/jfrog/log-analytics-prometheus/{{ .Values.global.jfrog.observability.branch }}/fluent_metrics.conf.rt -o {{ .Values.artifactory.persistence.mountPath }}/etc/fluentd/fluentd.conf
+      volumeMounts:
+        - mountPath: "{{ .Values.artifactory.persistence.mountPath }}"
+          name: volume
   customSidecarContainers: |
     - name: "artifactory-fluentd-sidecar"
-      image: "partnership-public-images.jfrog.io/fluentd/fluentd:1.18.0"
+      image: "releases-pts-observability-fluentd.jfrog.io/fluentd:2.1"
       imagePullPolicy: "IfNotPresent"
-      command:
-          - 'sh'
-          - '-c'
-          - >
-            gem install fluent-plugin-prometheus --no-doc;
-            curl https://raw.githubusercontent.com/jfrog/log-analytics-prometheus/{{ .Values.global.jfrog.observability.branch }}/fluent_metrics.conf.rt -o fluentd_metrics.conf;
-            cp fluentd_metrics.conf fluent.tmp.conf;
-            sed -i -e 's/LOKI_URL/{{ .Values.global.prometheus.loki_url }}/g' fluent.tmp.conf;
-            mv fluent.tmp.conf fluentd.conf;
-            fluentd -v -c fluentd.conf
       volumeMounts:
         - mountPath: "{{ .Values.artifactory.persistence.mountPath }}"
           name: volume
       env:
         - name: JF_PRODUCT_DATA_INTERNAL
           value: {{ .Values.artifactory.persistence.mountPath }}
+        - name: LOKI_URL
+          value: {{ .Values.global.prometheus.loki_url }}
+        - name: FLUENTD_CONF
+          value: ../../../../{{ .Values.artifactory.persistence.mountPath }}/etc/fluentd/fluentd.conf
```

Comment:
This may be problematic from security POV. Possible path traversal vulnerability here. Not sure if this can be avoided.

Reply:
Cannot be avoided because the path this variable takes is always read as /opt/bitnami/fluentd/conf/<path> in the initialize